### PR TITLE
New stubs: react-key-value

### PR DESCRIPTION
**Pull request by ckchow**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-key-value ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-key-value@latest/))
was missing the following properties:
1. KeyValue
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0401f87d805ab4b01a2ed971281693541464c18a
</p>

File: types/react-key-value/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+interface KeyValueProps {
+    rows?: Array<{keyItem: string, valueItem: string}>;
+    onChange?: (rows: Array<{keyItem: string, valueItem: string}>) => void;
+    customAddButtonRenderer?: (handleAddNew: () => any) => React.ReactElement;
+    keyInputPlaceholder?: string;
+    valueInputPlaceholder?: string;
+    hideLabels?: boolean;
+}
+
+export default KeyValue;
```
Comment (andrewbranch) [inline types/react-key-value/index.d.ts:40]:
```suggestion
export default KeyValue;
```

File: types/react-key-value/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    renderRows(): any;
+
+    renderValueItem(index: bigint, value: any): any;
+
+    toJSON(): any;
+
+    static displayName: string;
+}
+
+interface KeyValueProps {
```
Comment (andrewbranch) [inline types/react-key-value/index.d.ts:31]:
```suggestion
export interface KeyValueProps {
```

File: types/react-key-value/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for react-key-value 2.0
+// Project: https://github.com/purposeindustries/react-key-value#readme
+// Definitions by: Chris Chow <https://github.com/ckchow>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+import * as React from 'react';
+
+declare class KeyValue extends React.Component<KeyValueProps> {
```
Comment (andrewbranch) [inline types/react-key-value/index.d.ts:7]:
```suggestion
export declare class KeyValue extends React.Component<KeyValueProps> {
```

File: types/react-key-value/react-key-value-tests.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import KeyValue from 'react-key-value';
```
Comment (andrewbranch) [inline types/react-key-value/react-key-value-tests.tsx:1]:
```suggestion
import { KeyValue } from 'react-key-value';
```

**Review by andrewbranch — CHANGES_REQUESTED** [review]
The README on npm is wrong; check the README on GitHub, and the [actual source code](https://github.com/purposeindustries/react-key-value) (the best source of truth). `KeyValue` is a named export, not the default or `module.exports`.

**Comment by ckchow** [comment]
Thanks a bunch for the suggestions! I'm not super familiar with TS or react honestly so the explanation about the default export is appreciated.

**Comment by andrewbranch** [comment]
My inline suggestions weren’t quite good enough to pass the linter. Can you fix the errors locally?

**Review by andrewbranch — APPROVED** [review]

**Comment by ckchow** [comment]
Ready to merge

**Pull request merged**